testsupport/wait/host.go

```diff
@@ -63,20 +64,27 @@ func (a *HostAwaitility) WaitForMasterUserRecord(name string, criteria ...Master
 		obj := &toolchainv1alpha1.MasterUserRecord{}
 		if err := a.Client.Get(context.TODO(), types.NamespacedName{Namespace: a.Namespace, Name: name}, obj); err != nil {
 			if errors.IsNotFound(err) {
-				a.T.Logf("waiting for availability of MasterUserRecord '%s'", name)
 				return false, nil
 			}
 			return false, err
 		}
-		for _, match := range criteria {
-			if !match(a, obj) {
+		for _, c := range criteria {
+			if !c.Match(a, obj) {
 				return false, nil
 			}
 		}
-		a.T.Logf("found MasterUserRecord '%s': %+v with ClusterResources: %v", name, obj, obj.Spec.UserAccounts[0].Spec.NSTemplateSet.ClusterResources)
 		mur = obj
 		return true, nil
 	})
+	// log message if an error occurred
+	if err != nil {
+		a.T.Logf("failed to find MasterUserRecord '%s' with matching criteria: %v", name, err)
+		for _, c := range criteria {
+			if !c.Match(a, mur) {
+				a.T.Log(c.Diff(mur))
+			}
+		}
+	}
```

Comment:
it looks like that it could be the same for all wait functions - couldn't we introduce some more abstract wait framework that would do such operations automatically? You know - to avoid code duplications

Reply:
the `criteria` type may differ, so I'm not sure we can factorize much, but I will take a look

Reply:
done in a941d0c